File: FoodOn_backend/src/main/java/com/foodon/foodon/meal/dto/MealDetailInfoResponse.java
package com.foodon.foodon.meal.dto;

import com.foodon.foodon.meal.domain.Meal;
import com.foodon.foodon.meal.domain.MealTimeType;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

import static com.foodon.foodon.common.util.BigDecimalUtil.round;

public record MealDetailInfoResponse(
        Long mealId,
        String mealTime,
        MealTimeType mealTimeType,
        String imageFileName,
        BigDecimal totalKcal,
        BigDecimal totalCarbs,
        BigDecimal totalProtein,
        BigDecimal totalFat,
        List<MealItemInfo> mealItems
) {
    public static MealDetailInfoResponse from(
            Meal meal,
            List<MealItemInfo> mealItems
    ) {
        return new MealDetailInfoResponse(
                meal.getId(),
                meal.getMealTime().format(DateTimeFormatter.ofPattern("HH:mm")),
                meal.getMealTimeType(),
                meal.getMealImage(),
                round(meal.getTotalKcal(), 0),
                round(meal.getTotalCarbs(), 1),
                round(meal.getTotalProtein(), 1),
                round(meal.getTotalFat(), 1),
                mealItems
        );
    }
}

File: android-app/app/src/main/java/com/swallaby/foodon/data/food/remote/dto/response/MealDetailInfoResponse.kt
package com.swallaby.foodon.data.food.remote.dto.response

import androidx.core.net.toUri
import com.swallaby.foodon.domain.food.model.MealInfo
import com.swallaby.foodon.domain.food.model.MealItem
import com.swallaby.foodon.domain.food.model.MealType

data class MealDetailInfoResponse(
    val mealId: Long = 0,
    val mealDateTime: String = "",// LocalDateTime.now(),
    val mealTimeType: MealType = MealType.BREAKFAST,
    val imageFileName: String = "",
    val totalCarbs: Double = 0.0,
    val totalFat: Double = 0.0,
    val totalKcal: Double = 0.0,
    val totalProtein: Double = 0.0,
    val mealItems: List<MealItem> = emptyList(),
)

fun MealDetailInfoResponse.toDomain(): MealInfo = MealInfo(
    imageUri = imageFileName.toUri(),
    mealId = mealId,
    mealTime = mealDateTime,
    mealTimeType = mealTimeType,
    imageFileName = imageFileName,
    totalCarbs = totalCarbs,
    totalFat = totalFat,
    totalKcal = totalKcal.toInt(),
    totalProtein = totalProtein,
    mealItems = mealItems
)
